From 82b25b8facf5c7aeef57c67c45cbc1c90c8fa1b3 Mon Sep 17 00:00:00 2001
From: Miguel Martinez <slverwolf@gmail.com>
Date: Fri, 20 Apr 2018 08:50:16 -0500
Subject: [PATCH 1/1] Enable GSS garbage collection

GSS specifies the generational semi space plan. This plan is very similar to the semi-space plan with the exception that older-lived objects are promoted into a large RosAlloc space. This has the advantage of needing to copy significantly fewer objects for typical use cases.
---
 target/product/runtime_libart.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/target/product/runtime_libart.mk b/target/product/runtime_libart.mk
index f1b8f4596..21608e966 100644
--- a/target/product/runtime_libart.mk
+++ b/target/product/runtime_libart.mk
@@ -65,7 +65,8 @@ PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
     dalvik.vm.usejit=true \
     dalvik.vm.usejitprofiles=true \
     dalvik.vm.dexopt.secondary=true \
-    dalvik.vm.appimageformat=lz4
+    dalvik.vm.appimageformat=lz4 \
+    dalvik.vm.gctype=GSS
 
 # Different dexopt types for different package update/install times.
 # On eng builds, make "boot" reasons only extract for faster turnaround.
-- 
2.11.0

